tests/cpp-tests/Classes/UITest/CocoStudioGUITest/UITextFieldTest/UITextFieldTest.cpp

```diff
@@ -446,8 +446,8 @@ bool UITextFieldTest_BMFont::init()
         _uiLayer->addChild(alert);
         
         // Create the textfield
-        TextField* textField = TextField::create("input words here","fonts/markerFelt.fnt",25);
-        
+        TextField* textField = TextField::create("input words here","fonts/bitmapFontTest3.fnt",14);
+        textField->setCursorEnabled(true);
```

Comment:
did you find that the size is different between `BMFont Letter` and `Cursor` sometimes? ,it's visible when font size = 25

Reply:
Yeah, that was my issue. Don't know how to solve it, as it seems to behave differently on different fonts. When you open a .fnt file, you can see the font size at the top. Somehow it works for markerFelt but not for the other font. As if it does some internal scaling (or perhaps because of something with retina?)

Reply:
I hold the same idea with you, some internal scaling must exist. 

`typedef the old class to the new one` sounds good, it's way to solve compatibility. we can do it after we solve the blink issue on `BMFont` and `system font`.

Reply:
`void Label::updateLetterSpriteScale(Sprite* sprite)` this method can keep the _BMFont Cursor Sprite_ with other letters the same size, but when I try to add this into cursor update method, position isn't perfect. maybe we can find another place to put the scale method.

Reply:
I wil have a look later today. The cursor is blinking though, I fixed that.

Reply:
I found it! By default, a label will use the font size of the bmfont divided by `CC_CONTENT_SCALE_FACTOR()`. If you set a font size manually, which the widget does in this case (see TextField::create), the scale factor is not used. Perhaps we should add a different create method, just like Label has a Label::createWithBMFont method, which does not set the font size (as it will read it from the fnt file).
When I used the size 28/CC_CONTENT_SCALE_FACTOR(), it displayed correctly. So if you set a size manually and do a 'getLetter', it does not take the scale factor into account. Perhaps this is more a Label issue than a TextField issue.